modules/database/variables.tf

```diff
@@ -26,3 +26,25 @@ variable "deletion_protection" {
   type        = bool
   default     = true
 }
+
+
+variable "rds_db_retention_period" {
+  description = "The time between backups for the rds instance, default is disabled"
+  type        = number
+  default     = 0
+}
+
+variable "restore_to_point_in_time" {
```

Comment:
This looks good but it feels a bit weird to pass an array through here rather than an object.

e.g. rather than 
```hcl
restore_to_point_in_time = [{
          restore_time                  = "2024-04-01T16:05:00Z"
          source_db_instance_identifier = "prod-db"
          source_dbi_resource_id        = null
          use_latest_restorable_time    = null        
        }]
```

It would be preferable to do
```hcl
restore_to_point_in_time = {
          restore_time                  = "2024-04-01T16:05:00Z"
          source_db_instance_identifier = "prod-db"
          source_dbi_resource_id        = null
          use_latest_restorable_time    = null
        }
```

(unless there is a case when providing multiple `restore_to_point_in_time` entries is reasonable?)

Could you switch this to a nullable object instead? I think it should be possible following the guide below:

https://stackoverflow.com/questions/69373802/conditionally-execute-nested-block-restore-to-point-in-time

e.g.

```hcl
dynamic "restore_to_point_in_time" {
   for_each = var.restore_to_point_in_time != null ? [1] : []

   content {
    source_cluster_identifier         = var.restore_to_point_in_time.source_cluster_identifier
    ... rest of the variables here
    }
  }
```